src/observability/publisher_test.go

```diff
@@ -0,0 +1,115 @@
+package observability
+
+import (
+	"encoding/json"
+	"io"
+	"os"
+	"strings"
+	"testing"
+
+	logger "github.com/accuknox/auto-policy-discovery/src/logging"
+	"github.com/accuknox/auto-policy-discovery/src/types"
+	"github.com/stretchr/testify/assert"
+)
+
+var inputSummaryFileName = "input-summary.json"
+var aggSummaryFileName = "agg-summary.json"
+
+type SystemSummary struct {
+	SummaryData types.SystemSummary       `json:"summaryData"`
+	TimeCount   types.SysSummaryTimeCount `json:"timeCount"`
+}
+
+func makeSummaryMapFromTestCase(summaryData []SystemSummary, summaryMap map[types.SystemSummary]types.SysSummaryTimeCount) {
+
+	for _, ss := range summaryData {
+		summaryMap[ss.SummaryData] = ss.TimeCount
+	}
+
+}
+
+func initTestCase(t *testing.T, summaryMap, expectedSummaryMap map[types.SystemSummary]types.SysSummaryTimeCount) {
+	for expectedSummary, expectedSummaryCount := range expectedSummaryMap {
+		actualSummaryCount, ok := summaryMap[expectedSummary]
+		assert.True(t, ok, "Expected SummaryTimeCount not found in summaryMap")
+		assert.Equal(t, expectedSummaryCount.Count, actualSummaryCount.Count)
+		assert.Equal(t, expectedSummaryCount.UpdatedTime, actualSummaryCount.UpdatedTime)
+		log.Info().Msgf("Found:\n   PodName: %v,\n   Source: %v,\n   Destination: %v,\n   Operation: %v\n   Expected Count: %v\n   Actual Count: %v\n", expectedSummary.PodName, expectedSummary.Source, expectedSummary.Destination, expectedSummary.Operation, expectedSummaryCount.Count, actualSummaryCount.Count)
+		delete(expectedSummaryMap, expectedSummary)
+		delete(summaryMap, expectedSummary)
+	}
+}
+
+func Test_getAggregatedSummaryMap(t *testing.T) {
+
+	log = logger.GetInstance()
+	summaryMap := make(map[types.SystemSummary]types.SysSummaryTimeCount)
+	expectedSummaryMap := make(map[types.SystemSummary]types.SysSummaryTimeCount)
+
+	wd, err := os.Getwd()
+	if err == nil {
+		tempWD := strings.Split(wd, "/")
+		projectHome := strings.Join(tempWD[:len(tempWD)-2], "/")
+
+		inputSummaryFileName = projectHome + "/tests/observability/" + inputSummaryFileName
+		aggSummaryFileName = projectHome + "/tests/observability/" + aggSummaryFileName
+
+	}
```

Comment:
We can achieve the same result by simply using the relative paths in `inputSummaryFileName` & `aggSummaryFileName`. 

Both are effectively relative paths and going to break if we change the directory hierarchy.